roachpb/data.go

```diff
@@ -131,6 +132,8 @@ func (k EncodedKey) PrefixEnd() EncodedKey {
 }
 
 // Less compares two keys.
+// TODO(tschottdorf): see whether we actually needs this. We shouldn't be
```

Comment:
After some rabbit-holing there are 20 call sites left (down from 34). I think they're all redundant but they'll incur enough refactoring to warrant a new PR, so that's what I'd rather do.